ambry-account/src/main/java/com/github/ambry/account/AccountUtils.java

```diff
@@ -70,4 +73,95 @@ static boolean hasDuplicateAccountIdOrName(Collection<Account> accounts) {
     deprecatedContainers.addAll(accountService.getContainersByStatus(Container.ContainerStatus.INACTIVE));
     return deprecatedContainers;
   }
+
+  /**
+   * Compares and logs differences (if any) in Accounts.
+   * @param accountsInPrimary accounts in primary collection.
+   * @param accountsInSecondary accounts in secondary collection.
+   * @return true if accounts and their containers are same in both collections.
+   */
+  public static boolean compareAccounts(Collection<Account> accountsInPrimary,
+      Collection<Account> accountsInSecondary) {
+    boolean isSame = true;
+    Map<Short, Account> secondaryAccountMap = new HashMap<>();
+    accountsInSecondary.forEach(account -> secondaryAccountMap.put(account.getId(), account));
+
+    Set<Account> accountsMissingInSecondary = accountsInPrimary.stream()
+        .filter(account -> secondaryAccountMap.get(account.getId()) == null)
+        .collect(Collectors.toSet());
+
+    Set<Account> accountsDifferentInSecondary = new HashSet<>(accountsInPrimary);
+    accountsDifferentInSecondary.removeAll(accountsInSecondary);
+    accountsDifferentInSecondary.removeAll(accountsMissingInSecondary);
+
+    if (!accountsMissingInSecondary.isEmpty() || !accountsDifferentInSecondary.isEmpty()) {
+      isSame = false;
+
+      StringBuilder accountsInfo = new StringBuilder();
+
+      if (!accountsMissingInSecondary.isEmpty()) {
+        accountsInfo.append("[");
+        for (Account account : accountsMissingInSecondary) {
+          accountsInfo.append(AccountCollectionSerde.accountToJsonNoContainers(account).toString()).append(",");
+        }
+        accountsInfo.append("]");
+        logger.warn("Accounts found in primary and absent in secondary = {}", accountsInfo.toString());
```

Comment:
The method doesn't log if accounts/containers exist in secondary but not primary.  I suppose the assumption is that primary is updated first so this case is unlikely to happen?  That might be acceptable.